crates/runtime/src/debug_builtins.rs

```diff
@@ -0,0 +1,40 @@
+//! Debugger helpers.
+
+use crate::instance::InstanceHandle;
+use crate::vmcontext::VMContext;
+use wasmtime_environ::entity::EntityRef;
+use wasmtime_environ::wasm::MemoryIndex;
+
+static mut VMCTX_AND_MEMORY: (*mut VMContext, usize) = (std::ptr::null_mut(), 0);
+
+#[no_mangle]
+#[allow(dead_code, missing_docs)]
+pub unsafe extern "C" fn resolve_vmctx_memory_ptr(p: *const u32) -> *const u8 {
+    let ptr = std::ptr::read(p);
+    assert!(!VMCTX_AND_MEMORY.0.is_null());
+    let handle = InstanceHandle::from_vmctx(VMCTX_AND_MEMORY.0);
+    assert!(
+        VMCTX_AND_MEMORY.1 < handle.instance().module().local.memory_plans.len(),
+        "memory index for debugger is out of bounds"
+    );
+    let index = MemoryIndex::new(VMCTX_AND_MEMORY.1);
+    let mem = handle.instance().get_memory(index);
+    mem.base.add(ptr as usize)
+}
+
+#[no_mangle]
+#[allow(dead_code, missing_docs)]
+pub unsafe extern "C" fn set_vmctx_memory(vmctx_ptr: *mut VMContext) {
+    // TODO multi-memory
```

Comment:
Want to create a tracking issue for multi-memory and add a checkbox for this there? That way we're less likely to forget about this when we actually implement multi-memory.

Reply:
https://github.com/bytecodealliance/wasmtime/issues/1489